frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -108,9 +117,13 @@ export const HeadingWithAnchor: FunctionComponent<HeadingWithAnchorProps> = ({
     addScriptFinishedHandler,
     removeScriptFinishedHandler,
   } = React.useContext(AppContext)
-
   if (isSidebar) {
-    return React.createElement(tag, tagProps, children)
+    return (
+      // @ts-ignore
+      <StyledHeading as={tag} isCaption={false} isInSidebar={isSidebar}>
```

Comment:
@kmcgrady , should I remove this refactor or maybe add a ts-ignore. From what I tested, it seems to be fine but there seems to be a bug for using as and strings in the as prop. The type is a ReactElement and doesn't accept strings. 

Bug Reference: https://github.com/emotion-js/emotion/issues/2266

Reply:
I also tried using withComponent and that didn't end up working for me either.

Reply:
Wonder if we can do the following:

1. Create styled components for h1, h2, h3 (the ones we have defined in that switch statement).
2. We leave the decision of which component to use to be in the component itself rather than in the `StyledHeading`
3. No need to use `as` field you just use the correct _styled component_

This will satisfy the type system, not be a ton of logic and styled components simpler. Thoughts?

Reply:
I thought of that style but I decided not to go with it because I thought it would be a lot of reduplication of code. However, I am for it since it does satisfy the type system and makes understanding the code possibly simpler where it's easy to understand each tag returns a separate component.

Reply:
Looking at this further, it looks like the caption spacing was messed up because of this. `isCaption` is only referenced by the StreamlitMarkdown component and `HeadingWithAnchor` doesn't have access to such a prop. As a result, this caused some problems. For now, I will leave this unless I find something further but for now, will revert this refactoring.